### DISCUSSION TITLE
[Vision alignment]: Comparative benchmarking - high-speed

### DISCUSSION THREAD

**Discussion by joshbruce** [markedjs/marked#963] [General]
What other markdown compilers are there for JS now?

Would be worthwhile to put together a list of compilers (markdown-it, for example) and a complex (but sane) markdown file to do comparative benchmarking. 

If we aren't faster than them, then we aren't achieving the primary purpose of marked...high-speed.

See also #956

**Comment by worker8** [comment]
I found this site jsperf.com for comparing javascript library performance.

Someone wrote a test for comparing the performance for a list of markdown compilers:

https://jsperf.com/showdown-js-vs-marked/10

![screen shot 2018-01-05 at 12 39 29 pm](https://user-images.githubusercontent.com/1988156/34594626-8ddf4b74-f215-11e7-9257-7bedf2245c56.png)

According to the tests, there are a few libraries ranked higher than marked. But without comparing the features of each markdown libs provide, this might not be a fair comparison... Like if they don't provide syntax highlight then for sure it will be faster.

This table can however be a starting point to learn about other markdown libraries.

**Comment by Feder1co5oave** [comment]
newer version:
https://jsperf.com/showdown-js-vs-marked/19
also:
https://github.com/commonmark/commonmark.js#performance

**Comment by calculuschild** [comment]
Two things. First, is there a good online benchmark anymore? The jsperf links above seem to be broken.

Second, running the Marked bench script on my own machine is showing Commonmark and Marked-It as both faster and more spec compliant. Can you point to any techniques that used to give you the edge that aren't working anymore, or recent updates that introduced a lot of slowdown? I'd be interested in taking a look.

**Comment by UziTech** [comment]
I don't know of any online benchmarks.

I think #1686 made a bit of a slow down but also markdown-it has been getting faster with recent releases. I don't know of any simple ways to increase the speed.

**Comment by calculuschild** [comment]
> I think #1686 made a bit of a slow down

I'm sure it did, there was a lot going on in there. I wonder if... there's a way to show kind of a benchmark score per release? Run some kind of automated benchmark on all releases up to this point, and collect that info somewhere useful so it could be plotted and compared, maybe as a page on the documentation?

**Comment by UziTech** [comment]
We tried to integrate the benchmarks into our testing but test servers are extremely inconsistent

**Comment by asbjornu** [comment]
[This MeasureThat benchmark](https://www.measurethat.net/Benchmarks/Show/32365/0/markdown-performance-comparison-2024-10-23) works (tested in Firefox 131.0.3 on macOS Sonoma 14.7 on a MacBook Pro M2):

<img width="1158" alt="JavaScript Markdown library benchmark results. Remarkable is fastest." src="https://github.com/user-attachments/assets/e58ab186-9d92-4c20-a490-39b6f2043c61">

The results seem to be rather consistent with Remarkable being the fastest and Remark the slowest, which is a bit surprising given https://github.com/pngwn/MDsveX/issues/20#issue-490629016, but the benchmark may just be using Remark wrong. Remarked ends up in 5th place.
